addons/survey/models/res_lang.py

```diff
@@ -0,0 +1,26 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command, models, _
+from odoo.exceptions import UserError
+
+
+class ResLang(models.Model):
+    _inherit = "res.lang"
+
+    def write(self, vals):
+        """ When a language is disabled, clear corresponding survey language. """
+        if 'active' in vals and not vals['active']:
+            self.env['survey.user_input'].search([('lang_id', 'in', self.ids)]).lang_id = False
+            surveys_sudo = self.env['survey.survey'].search([('lang_ids', 'in', self.ids)]).sudo()
```

Comment:
Or you don't find all records, do you?
```suggestion
            surveys_sudo = self.env['survey.survey'].sudo().search([('lang_ids', 'in', self.ids)])
```